### ISSUE TITLE
IE 11: Unable to get value of the property 'firstChild': object is null or undefined

### ISSUE THREAD

**Issue by gorshkov-leonid** [webcomponents/html-imports#22]
```
  //@webcomponents\html-imports\src\html-imports.js
  //    makeDocument(resource, url) {
  const template = /** @type {!HTMLTemplateElement} */
        (document.createElement('template'));
       template.innerHTML = resource; <------ template is HTMLUnknownElement 
```

```
  //@webcomponents\shadydom\src\patch-accessors.js
  innerHTML: {
    get() {
      ...
    },
    set(text) {
      let content = this.localName === 'template' ? this.content : this; <-- localName of  HTMLUnknownElement is 'template', but this.content == undefined
      clearNode(content); <------
      ...
    }

  function clearNode(node) {
    while (node.firstChild) { <-- node == undefined
      node.removeChild(node.firstChild);
    }
  }
````

It is bug?

**Comment by valdrinkoshi** [comment]
Hi @gorshkov-leonid, seems like an issue with the `template + shadydom` polyfills rather than `html-imports`...Which versions of those polyfills are you testing? Tried with `#master` for both and it seems to work fine in IE11 http://jsbin.com/jahukov/3/edit?html,output

**Comment by gorshkov-leonid** [comment]
Hi @valdrinkoshi, I committed example to branch https://github.com/gorshkov-leonid/html_imports_ISSUE_22 as is. 
I used dependencies from last commits of master branches.
```
    "@webcomponents/html-imports": "https://github.com/webcomponents/html-imports/archive/2a0b016.tar.gz",
    "@webcomponents/custom-elements": "https://github.com/webcomponents/custom-elements/archive/36451e1.tar.gz",
    "@webcomponents/shadycss": "https://github.com/webcomponents/shadycss/archive/bea402d.tar.gz",
    "@webcomponents/shadydom": "https://github.com/webcomponents/shadydom/archive/50bcf0a.tar.gz",
    "array.from": "^1.0.3",
    "es6-promise": "^4.0.5",
    "object.assign": "^4.0.4"
```

**Comment by gorshkov-leonid** [comment]
Oh yeah, I see https://github.com/webcomponents/template. May be need add references to shadydom's readme or somewhere else.
But I see new error SCRIPT1010: Expected identifier at `function h(Ht,...jt)` (shadydom). It is not es5?

**Comment by gorshkov-leonid** [comment]
When I recopiled shadydom via 'gulp es5' this example failed in FF and still not working in IE. Although  error log is clear in IE

**Comment by valdrinkoshi** [comment]
Looks like an error in the order of which you import the scripts in the page...Template script should be loaded before the others, then HTMLImports, then shadydom and finally customelements. See the gulp-all-the-things branch in webcomponentsjs for a reference (we are still working on it, it's WIP)

**Comment by gorshkov-leonid** [comment]
Error occured in 
```
//node_modules\@webcomponents\shadydom\src\patch-events.js
function mixinComposedFlag(Base) {
  ...  
  let klazz = function(type, options) {
    let event = new Base(type, options); // <------ 'Object doesn't support this action'
    ....
    return event;
  };
  ...
}
```
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/105506/
https://github.com/sindresorhus/devtools-detect/issues/9

I think, error logs are clear because somewhere was not called catch on promise

**Comment by gorshkov-leonid** [comment]
link to https://github.com/webcomponents/shadydom/pull/22 created by mistake

**Comment by valdrinkoshi** [comment]
Please check this entrypoint https://github.com/webcomponents/webcomponentsjs/blob/gulp-all-the-things/entrypoints/webcomponents-hi-sd-ce-pf-index.js#L17
IE11 needs to polyfill a couple things, and the error you stumbled upon is because of the missing polyfill for CustomEvent (which is done by https://github.com/webcomponents/webcomponents-platform/)

**Comment by gorshkov-leonid** [comment]
Thank you very much, it is works.  But I spent a lot of time to research the problem. May be better describe all steps in 'How to' format for v1. I see that old and new dependencies are shuffled and I don't understand which one is need. 
```
"webcomponents-platform", "template",  "@webcomponents/html-imports", "@webcomponents/custom-elements", "@webcomponents/shadydom", "@webcomponents/shadycss",
```
And how about to place all of them to @webcomponents?

And there are 2 problems still remain: 
1. Minified shadydom does not work in FF (tested in 51.0.1) and IE 11 
2. Errors is are swallowed in when in promise usages (html-imports or shadydom?)
I should to create a new ticket?

**Comment by valdrinkoshi** [comment]
We're working on this as we speak, this is all work in progress 😉 
We'll provide the needed documentation once we reach stability (we're testing all the chain up to polymer 👌).
We're experiencing the same frustration regarding errors swallowed by promises (`html-imports` just got a fix merged regarding this #18).
My suggestion is (for the time being) to follow the progress on https://github.com/webcomponents/webcomponentsjs/pull/661 regarding the "right" ordering of these polyfills 😉

**Comment by gorshkov-leonid** [comment]
Thank you very match for the fast explaination! I will follow #661. Really looking forward to the release.
